src/app/modules/item/pages/item-forum/item-forum.component.ts

```diff
@@ -0,0 +1,172 @@
+import { ReplaySubject, switchMap, combineLatest, Subject, first } from 'rxjs';
+import { mapToFetchState } from '../../../../shared/operators/state';
+import { distinctUntilChanged, filter, map, startWith, withLatestFrom } from 'rxjs/operators';
+import { ItemRouter } from '../../../../shared/routing/item-router';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { ItemType } from '../../../../shared/helpers/item-type';
+import { Item } from '../../http-services/get-item-by-id.service';
+
+interface Column {
+  field: string,
+  header: string,
+}
+
+enum ForumTabUrls {
+  MyThreads= '/forum/my-threads',
+  Others = '/forum/others',
+  Group = '/forum/group',
+}
+
+const OPTIONS = [
+  {
+    label: $localize`My threads`,
+    value: ForumTabUrls.MyThreads,
+  },
+  {
+    label: $localize`Other's`,
+    value: ForumTabUrls.Others,
+  },
+];
+
+@Component({
+  selector: 'alg-item-forum',
+  templateUrl: './item-forum.component.html',
+  styleUrls: [ './item-forum.component.scss' ],
+})
+export class ItemForumComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() itemData?: ItemData;
+
+  private readonly url$ = this.router.events.pipe(
+    filter(event => event instanceof NavigationEnd),
+    map(() => this.router.url),
+    startWith(this.router.url),
+    distinctUntilChanged(),
+  );
+  private readonly refresh$ = new Subject<void>();
+  private readonly item$ = new ReplaySubject<Item>(1);
+  private readonly watchedGroup$ = this.groupWatchingService.watchedGroup$;
+  private readonly destroyed$ = new Subject<void>();
+  isWatching$ = this.groupWatchingService.isWatching$;
+  selected$ = new ReplaySubject<number>(1);
+  options$ = combineLatest([
+    this.watchedGroup$,
+    this.url$,
+  ]).pipe(
+    map(([ watchedGroup, url ]) => [
+      ...OPTIONS,
+      ...(watchedGroup || url.endsWith(ForumTabUrls.Group)
+        ? [{ label: `${ watchedGroup?.name || $localize`Group` }'s`, value: ForumTabUrls.Group }] : [])
+    ]),
+  );
+  state$ = combineLatest([
+    this.selected$,
+    this.item$,
+    this.watchedGroup$,
+  ]).pipe(
+    switchMap(([ selected, item, watchedGroup ]) =>
+      this.getThreadService.get(selected === 2 && watchedGroup
+        ? { itemId: item.id, watchedGroupId: watchedGroup.route.id }
+        : { itemId: item.id, isMine: selected === 0 }).pipe(
+        map(threads => ({
+          columns: this.getThreadColumns(item.type),
+          rowData: threads,
+        })),
+        mapToFetchState({ resetter: this.refresh$ }),
+      ),
+    ),
+  );
+
+  constructor(
+    private groupWatchingService: GroupWatchingService,
+    private getThreadService: GetThreadsService,
+    private itemRouter: ItemRouter,
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+  ) {
+  }
+
+  ngOnChanges(): void {
+    if (this.itemData) {
+      this.item$.next(this.itemData.item);
+    }
+  }
+
+  ngOnInit(): void {
+    this.url$.pipe(
```

Comment:
should be moved to a subscription var which is unsubscribed at the end

Reply:
I suppose that the `first` will do this work, from rxjs doc:

If called with no arguments, `first` emits the first value of the source Observable, then completes.

Reply:
Right!